aws/kms/variables.tf

```diff
@@ -0,0 +1,52 @@
+variable "name" {
+  description = "The name of the key as viewed in AWS console."
+  type        = string
+}
+
+variable "description" {
+  description = "The description of the key as viewed in AWS console."
+  type        = string
+  default     = null
+}
+
+variable "deletion_window_in_days" {
+  description = "The waiting period, specified in number of days."
+  type        = number
+  default     = 7
+}
+
+variable "key_usage" {
+  description = "Specifies the intended use of the key. Valid values: ENCRYPT_DECRYPT, SIGN_VERIFY, or GENERATE_VERIFY_MAC."
+  type        = string
+  default     = "ENCRYPT_DECRYPT"
+}
+
+variable "custom_key_store_id" {
+  description = "ID of the KMS Custom Key Store where the key will be stored instead of KMS (eg CloudHSM)."
+  type        = string
+  default     = null
+}
+
+variable "customer_master_key_spec" {
+  description = "Specifies whether the key contains a symmetric key or an asymmetric key pair and the encryption algorithms or signing algorithms that the key supports. Valid values: SYMMETRIC_DEFAULT, RSA_2048, RSA_3072, RSA_4096, HMAC_256, ECC_NIST_P256, ECC_NIST_P384, ECC_NIST_P521, or ECC_SECG_P256K1."
+  type        = string
+  default     = "SYMMETRIC_DEFAULT"
+}
+
+variable "enable_key_rotation" {
+  description = "Specifies whether key rotation is enabled. Required to be enabled if rotation_period_in_days is specified."
+  type        = bool
+  default     = false
+}
+
+variable "rotation_period_in_days" {
+  description = "Custom period of time between each rotation date. Must be a number between 90 and 2560."
+  type        = number
+  default     = 90
+}
+
+variable "enabled_create_policy" {
+  description = "Specifies whether policy need to be created."
+  type        = bool
+  default     = false
+}
```

Comment:
I think we should remove this variable and always create corresponding IAM to the resource. Can you share more of your thoughts on this?

Reply:
I think in the future I would add an option to attach a custom policy instead of default creation. Therefore, I created a variable to handle it.